planner/funcdep/fd_graph.go

```diff
@@ -538,3 +900,24 @@ func (e *fdEdge) String() string {
 	}
 	return b.String()
 }
+
+// RegisterUniqueID is used to record the map relationship between expr and allocated uniqueID.
+func (s *FDSet) RegisterUniqueID(hashCode string, uniqueID int) {
+	if len(hashCode) == 0 {
+		// shouldn't be here.
+		panic("map empty expr hashcode to uniqueID")
+	}
+	if _, ok := s.HashCodeToUniqueID[hashCode]; ok {
+		// shouldn't be here.
+		panic("hashcode has been registered")
```

Comment:
I think using `panic` is not safe, error with appropriate handling is preferred. There are 5 places in this PR where `panic` appears.

Reply:
Use log instead